Update PortalFetch_download

uniquePaul opened the pull request:
First pull request from Paul

Commit:
Update CODEOWNERS

yifeili98 commented:
I think you should either put ```@uniquePaul``` or simply ```ypwen@ucdavis.edu```. Not sure if ```@ypwen@ucdavis.edu``` works

uniquePaul commented:
> First pull request from Paul

Oh, Thanks

> I think you should either put `@uniquePaul` or simply `ypwen@ucdavis.edu`. Not sure if `@ypwen@ucdavis.edu` works

hk-mp5a3 commented on .github/CODEOWNERS:
```diff
@@ -6,4 +6,4 @@
 # review when someone opens a pull request.
-*   @kis87988 @ak840206 @AmberWow123 @biankaiwen111 @emilylxf @zhirezhixin @FHDA/FHDATime-Team
\ No newline at end of file
+*   @kis87988 @ak840206 @AmberWow123 @biankaiwen111 @emilylxf @zhirezhixin @FHDA/FHDATime-Team @ypwen@ucdavis.edu
```
maybe you should put `@uniquePaul` here

yifeili98 commented on .github/CODEOWNERS:
```diff
@@ -6,4 +6,4 @@
 # the repo. Unless a later match takes precedence,
 # @global-owner1 and @global-owner2 will be requested for
 # review when someone opens a pull request.
-*   @kis87988 @ak840206 @AmberWow123 @biankaiwen111 @emilylxf @zhirezhixin @FHDA/FHDATime-Team
```
顺便帮我把```@zhirezhixin```改成```@yifeili98```吧
阿里嘎多！

Commit:
Update CODEOWNERS

yifeili98 reviewed (approved):
I think your user name is uniquePaul so you should put ```@uniquePaul``` here.  

However there is also a ```@FHDA/FHDATime-Team``` which is a team you are already in, so I will give this one a pass.  

I also plan to remove everyone and leave this line with only a single ```@FHDA/FHDATime-Team``` so we only need to modify the team instead of this file in the future.

hk-mp5a3 commented:
> I think your user name is uniquePaul so you should put `@uniquePaul` here.
> 
> However there is also a `@FHDA/FHDATime-Team` which is a team you are already in, so I will give this one a pass.
> 
> I also plan to remove everyone and leave this line with only a single `@FHDA/FHDATime-Team` so we only need to modify the team instead of this file in the future.

Yifei, I think `@Paulwen` should not be added here as it indicates another github user: https://github.com/Paulwen and therefore it could cause an issue. We should restrict the code owner to `@FHDA/FHDATime-Team` or only add `@uniquePaul`

hk-mp5a3 commented:
Essentially, the `CODEOWNERS` file is used to "define individuals or teams that are responsible for code in a repository".
And "The pattern is followed by one or more GitHub usernames or team names using the standard `@username` or `@org/team-name` format". In this context, only your username should be added to the `CODEOWNERS` file.

Docs:
[1] https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

yifeili98 reviewed (changes requested):
LOL nope... my bad... See comment above...🤣

Commit:
refactor: add func to optimize input

Using the quarter_value to generate quarter and filename

yifeili98 commented on PortalFetch/crawler_download.py:
```diff
@@ -204,6 +206,38 @@ def waitUtilPageLoaded(driver, count):
+    year = quarterValue[0:4]
+    resultList = []
+    quarterSwitcher = {
+        "1": "Summer",
+        "2": "Fall",
+        "3": "Winter",
+        "4": "Spring",
```
nit: please be consistent with using single quotes or double quotes for the entire project

yifeili98 commented on PortalFetch/crawler_download.py:
```diff
@@ -204,6 +206,38 @@ def waitUtilPageLoaded(driver, count):
+        school = "De_Anza"
+    fn = year + "_" + quarter + "_" + school + "_courseData.json"
+    resultList.append(fn)
+    return resultList
```
the option value in the dropdown list is a bit tricky, they use summer as the start of a new academic year. This implies that the summer quarter of 202x is actually the first quarter of 202(x+1), so you will need to change some code logic here so that **if the input value is ```202112```, the filename and first key inside should correspond to summer 2020 instead of summer 2021**

yifeili98 commented on PortalFetch/crawler_download.py:
```diff
@@ -204,6 +206,38 @@ def waitUtilPageLoaded(driver, count):
+    }
+    school = schoolSwitcher.get(quarterValue[5], "")
+    quarter = quarterSwitcher.get(quarterValue[4], "")
+    qr = year + " " + quarter + " " + school
```
nit: please come up with some meaningful variable names

hk-mp5a3 replied:
change `qr` to a meaningful name

yifeili98 commented on PortalFetch/crawler_download.py:
```diff
@@ -204,6 +206,38 @@ def waitUtilPageLoaded(driver, count):
+    resultList.append(qr)
+    if school == "De Anza":
+        school = "De_Anza"
+    fn = year + "_" + quarter + "_" + school + "_courseData.json"
```
nit: please come up with some meaningful variable names

hk-mp5a3 replied:
change "fn" to a meaningful name. 

Readings:
[Clean code 101 — Meaningful names and functions](https://medium.com/coding-skills/clean-code-101-meaningful-names-and-functions-bf450456d90c#:~:text=How%20meaningful%20%3F,does%20not%20reveal%20its%20intent.)

hk-mp5a3 commented on PortalFetch/crawler_download.py:
```diff
@@ -244,8 +278,10 @@ def main():
         html = saveResult(driver)
-        filename = parser.get('config', 'filename')
-        quarter = parser.get('config', 'quarter')
+        # get Quarter and fileName based On quarter_value in crawler.config
```
Should be changed to `get quarter and filename based on quarter_value in crawler.config`.

hk-mp5a3 commented on PortalFetch/crawler_download.py:
```diff
@@ -244,8 +278,10 @@ def main():
-        quarter = parser.get('config', 'quarter')
+        # get Quarter and fileName based On quarter_value in crawler.config
+        filename = generateFilenameAndQuarter(value)[1]
+        quarter = generateFilenameAndQuarter(value)[0]
```
You should use `quarter, filename = generateFilenameAndQuarter(value)`

In this case, you won't call the same function for two times. This will reduce the cost of the code.

hk-mp5a3 commented on .github/CODEOWNERS:
```diff
@@ -6,4 +6,4 @@
 # review when someone opens a pull request.
-*   @kis87988 @ak840206 @AmberWow123 @biankaiwen111 @emilylxf @zhirezhixin @FHDA/FHDATime-Team
\ No newline at end of file
+*   @kis87988 @ak840206 @AmberWow123 @biankaiwen111 @emilylxf @zhirezhixin @FHDA/FHDATime-Team @Paulwen
```
Again, please change `@Paulwen` to `@uniquePaul`

hk-mp5a3 commented:
If you are trying to do something other than `Update CODEOWNERS`, please change your pull request description. 
Usually, it would be better to split mixed changes into small single pull requests.

Commit:
refactor: add func to optimize input

Updating logic of Summer quarter, fixing meaningless var names, Calling func only once

hk-mp5a3 commented on PortalFetch/crawler_download.py:
```diff
@@ -244,8 +280,11 @@ def main():
-        filename = parser.get('config', 'filename')
-        quarter = parser.get('config', 'quarter')
+        # get quarter and fileName based On quarter_value in crawler.config
+        resultList = generateFilenameAndQuarter(value)
```
1. The `generateFilenameAndQuarter` function should be rename to `generateQuarterAndFilename` since quarter is your first return value, and filename is the second return value. This will make your code more readable.
2. I would avoid having
```
resultList = generateFilenameAndQuarter(value)
filename = resultList[1]
quarter = resultList[0]
```
It is better to do 
```
quarter, filename = GenerateQuarterAndFilename(value)
```

hk-mp5a3 commented on PortalFetch/crawler_download.py:
```diff
@@ -244,8 +280,11 @@ def main():
         html = saveResult(driver)
-        filename = parser.get('config', 'filename')
-        quarter = parser.get('config', 'quarter')
+        # get quarter and fileName based On quarter_value in crawler.config
```
As mentioned, this should be changed to `get quarter and filename based on quarter_value in crawler.config`.
`fileName` -> `filename` because you are using `filename` as variable name in this function
`On` -> `on` because `O` should not be upper case in this context

hk-mp5a3 commented on PortalFetch/crawler_download.py:
```diff
@@ -204,6 +206,40 @@ def waitUtilPageLoaded(driver, count):
     raise ElementNotVisibleException("Could not load the full page!")
 
 
+def generateFilenameAndQuarter(quarterValue):
```
rename to `generateQuarterAndFilename`.

hk-mp5a3 commented on PortalFetch/crawler_download.py:
```diff
@@ -204,6 +206,40 @@ def waitUtilPageLoaded(driver, count):
+        school = "De_Anza"
+    fileNameOutput = year + "_" + quarter + "_" + school + "_courseData.json"
+    resultList.append(fileNameOutput)
+    return resultList
```
instead of having 
```
resultList.append(quarterOutput)
resultList.append(fileNameOutput)
return resultList
```
You can just 
```
return quarterOutput, fileNameOutput
```

Commit:
refactor: fix problems

returning two vars from generateQuarterAndFilename, changed/refined comments

hk-mp5a3 reviewed (approved):
LGTM. Thanks for the changes

yifeili98 reviewed (approved):
LGTM
